src/d3a/models/read_user_profile.py

```diff
@@ -53,26 +55,20 @@ def _str_to_datetime(time_str, time_format) -> DateTime:
 
 def default_profile_dict(val=None) -> Dict[DateTime, int]:
     """
-    Expanding a dictionary that contains one key for every minute of the simulation time
-    The keys are pendulum (DateTime) objects
-    :return: Dict[DateTime, int]
+    Returns dictionary with default values for all market slots.
+    :param val: Default value
     """
     if val is None:
         val = 0
+    if IS_CANARY_NETWORK:
+        market_slot_list = \
+            generate_market_slot_list(start_date=today(tz=TIME_ZONE),
+                                      time_span=duration(
+                                          days=CN_PROFILE_EXPANSION_DAYS))
```

Comment:
Im wondering wouldn't it be good to implement via skipping datetime/pendulum object to make d3a as light weight as possible for CN?

Reply:
You mean instead of using datetime/pendulum to use unix-timestamp? 
This will be done in the future probably. But for now, as the profiles are rotated and we are not going into the BC soon, pendulum is too handy to be refactored. Just think about to reimplement something like `weekday()` in unix timestamps.
It is really not that heavy, even if we have 1000 communities (from the memory points of view). For the CPU it really is not big of a deal as we run in real time. 
I hope this answers you question.